### PR TITLE
Add support to configure host kernel

### PR DESCRIPTION
**Pull request by eliud-kyale**
This commits adds support for host kernel configurations new field in hostprofilespec 'kernel'

Test plan:

Testing (starlingx AIO-SX)

- update kernel parameter in deployment-config.yaml
- kubectl apply -f deployment-config.yaml
- verfiy new kernel values system host-kernel-show controller-0
- test AIO-DX and AIO-SX

### CODE REVIEW

**Review by yjian118 — CHANGES_REQUESTED** [review]
Needs additional change in host controller to reconcile the new spec. After the change, please try "make" to create new DM image and helm chart, the helm chart should also be included in the commit

**Comment by eliud-kyale** [comment]
> Needs additional change in host controller to reconcile the new spec. After the change, please try "make" to create new DM image and helm chart, the helm chart should also be included in the commit

Add the DM image and helm chart ( I think )

**Comment by eliud-kyale** [comment]
### Test passed .

- deployctl updates the kernel value in deployment-config.yaml 
```
kind: HostProfile
metadata:
  labels:
    controller-tools.k8s.io: "1.0"
  name: controller-0-profile
  namespace: deployment
spec:
  administrativeState: unlocked
  appArmor: disabled
  boardManagement:
    type: none
  bootDevice: /dev/disk/by-path/pci-0000:00:01.1-ata-1.0
  clockSynchronization: ntp
  console: ttyS0,115200n8
  hwSettle: "0"
  installOutput: graphical
  interfaces:
    ethernet:
    - class: platform
      dataNetworks: []
      name: oam0
      platformNetworks:
      - oam
      port:
        name: enp0s3
      ptpInterfaces: []
      ptpRole: none
      uuid: ""
  kernel: **standard**
```
### After modifying kernel : lowlatency

- kubectl apply sets the new kernel value 

standard -> lowlatency 

[sysadmin@controller-0 ~(keystone_admin)]$ system host-kernel-show controller-0
+--------------------+--------------+
| Property           | Value        |
+--------------------+--------------+
| hostname           | controller-0 |
| kernel_provisioned | lowlatency   |
| kernel_running     | lowlatency   |
+--------------------+--------------+

File: go.mod
Comment (yjian118) [inline go.mod:94]: Need to switch back to the windriver repo after the commit in the gophercloud merged

**Review by yjian118 — CHANGES_REQUESTED** [review]
Hold this PR until switching back to the wind-river/gophercloud

File: controllers/host/host_controller.go
Comment (eliud-kyale) [inline controllers/host/host_controller.go:252]: the subfunctions can't be modified after initial provisioning . 
And if they are set by the install menus anyway 

host-update doesn't need the subfunctions to be specified
info is recieved from /etc/platform/platform.conf 

```
sysadmin@controller-0:~$ cat /etc/platform/platform.conf
nodetype=controller
subfunction=controller,worker
system_type=All-in-one
http_port=8080
management_interface=enp0s8
INSTALL_UUID=16a32937-2c2c-49e5-8e9f-295ade9a83e3
UUID=4c57a190-dea5-40e2-9469-2f9123def582
sw_version=23.09
vswitch_type=none
system_mode=duplex
sdn_enabled=no
region_config=no
security_feature="nopti nospectre_v2 nospectre_v1"
cluster_host_interface=enp0s8
oam_interface=enp0s3

```

Comment (eliud-kyale) [reply controllers/host/host_controller.go:252]: After removing the subfunction reconcile code the kernel changes can be modified 
standard -> lowlatency -> standard without sysinv rest api rejecting the subfunction patch

**Comment by eliud-kyale** [comment]
Test AIO-DX 
```

[sysadmin@controller-0 ~(keystone_admin)]$ kubectl -n deployment get datanetworks,hostprofiles,hosts,platformnetworks,systems
NAME                                                       BASE
hostprofile.starlingx.windriver.com/controller-0-profile
hostprofile.starlingx.windriver.com/controller-1-profile

NAME                                        ADMINISTRATIVE   OPERATIONAL   AVAILABILITY   PROFILE                INSYNC   SCOPE       RECONCILED
host.starlingx.windriver.com/controller-0   unlocked         enabled       available      controller-0-profile   true     bootstrap   true
host.starlingx.windriver.com/controller-1   unlocked         enabled       available      controller-1-profile   true     bootstrap   true

NAME                                  MODE     TYPE         VERSION   INSYNC   SCOPE       RECONCILED
system.starlingx.windriver.com/vbox   duplex   all-in-one   23.09     true     bootstrap   true
[sysadmin@controller-0 ~(keystone_admin)]$ system host-kernel-show controller-0
+--------------------+--------------+
| Property           | Value        |
+--------------------+--------------+
| hostname           | controller-0 |
| kernel_provisioned | lowlatency   |
| kernel_running     | lowlatency   |
+--------------------+--------------+
[sysadmin@controller-0 ~(keystone_admin)]$ system host-kernel-show controller-1
+--------------------+--------------+
| Property           | Value        |
+--------------------+--------------+
| hostname           | controller-1 |
| kernel_provisioned | lowlatency   |
| kernel_running     | lowlatency   |
+--------------------+--------------+

```

File: controllers/host/host_controller.go
Comment (ttakenak) [reply controllers/host/host_controller.go:252]: Removing this part means we don't support subfunction configuration in bootstrap via configuration yaml?
Also what is the insync status after bootstrap and day2 operation? Does it become true?

Comment (eliud-kyale) [reply controllers/host/host_controller.go:252]: After our online discussion. I was goind to remove subfunctions from the CRDs but the "worker" subfunction is used everywhere in the logic. I will attempt to create some if the logic around lowlatency and the kernel

**Review by yjian118 — CHANGES_REQUESTED** [review]
Expect two test cases:
1 deploy a system without the lowlatency in subfunctions but have kernel as lowlatency
2 deploy a system with the lowlatency in subfunctions but with the kernel as standard,

I'd like you to expect results about the INSYNC, RECONCILED status, and the subfunctions generated by the deployctl. You'll be able to update the result in the PR conversation.

**Review by yjian118 — COMMENTED** [review]
Expect unit tests to cover new functions

**Comment by eliud-kyale** [comment]
Unit tests added and passing

**Review by yjian118 — APPROVED** [review]

**Comment by ttakenak** [comment]
Could you add the test case described above:
- deploy a system with the lowlatency in subfunctions but with the kernel as standard,

**Comment by eliud-kyale** [comment]
Test 1: (AIO_DX)
  - kernel : lowlatency
  - subfunctions : [ controller, worker] 
  
  Test 2: (AIO-DX)
 - kernel : standard 
 - subfunctions : [ controller, worker, lowlatency ] 
 
 Both passing 
```  
[sysadmin@controller-0 ~(keystone_admin)]$ kubectl -n deployment get datanetworks,hostprofiles,hosts,platformnetworks,systems
NAME                                                       BASE
hostprofile.starlingx.windriver.com/controller-0-profile
hostprofile.starlingx.windriver.com/controller-1-profile

NAME                                        ADMINISTRATIVE   OPERATIONAL   AVAILABILITY   PROFILE                INSYNC   SCOPE       RECONCILED
host.starlingx.windriver.com/controller-0   unlocked         enabled       available      controller-0-profile   true     bootstrap   true
host.starlingx.windriver.com/controller-1   unlocked         enabled       available      controller-1-profile   true     bootstrap   true

NAME                                  MODE     TYPE         VERSION   INSYNC   SCOPE       RECONCILED
system.starlingx.windriver.com/vbox   duplex   all-in-one   23.09     true     bootstrap   true
```

**Review by ttakenak — APPROVED** [review]

**Pull request merged**
